### ISSUE TITLE
MacOS Un-Mute doesn't work (when using headphones)

### ISSUE THREAD

**Issue by thisIsTheFoxe** [nextcloud/spreed#6570]
## Steps to reproduce

It's pretty unstable and not easy to reproduce, since it'd be different every time, especially when having Safari Settings -> Websites -> Microphone is set to "Ask" (default). When "Allow"ing mic access by default for the webpage, it seems to happen less frequently.

That said, the most reproducible example I found is:

1. Use wireless headphones (e.g. AirPods) as output device
2. Open the talk page in Safari in a new window
3. Join call, get pop up "Camera and microphone setup"
4. _Quickly_ "Join call" (before both (cam & mic) icons are loaded & enabled)
5. Mute yourself
6. Unmute

### Expected behaviour
Mic unmutes

### Actual behaviour
Other participants can't hear me. As far as I could tell, it's still active at this point, it just doesn't send anything. Now, when just reloading the page the mic doesn't work at all anymore. The only way to fix it is to close the entire window / tab and reopen it in a new one. 
Sometimes, switching input devices helps, but often it then tries to reload the call and disconnects.
Other times, the mic would not work right after step 3.

Waiting for a bit after "Camera and microphone setup" is shown makes it happen less frequently.

## Browser

**Microphone available:** yes

**Camera available:** yes

**Operating system:** macOS Monterey 12.0.1 (21A559)

**Browser:** Safari 15.1 (17612.2.9.1.20)

### Browser log

```
[Error] Failed to create MediaStream audio source: Request is not allowed (x2)
[Error] Error getting audio stream: NotReadableError: The I/O read operation failed.
	(anonymous function) (talk-main.js:1:7086944)
	promiseReactionJob



[Error] Unhandled Promise Rejection: NotSupportedError: The operation is not supported.
	(anonymous function)
	rejectPromise


    [Error] Failed to create MediaStream audio source: Request is not allowed (x2)
[Error] Error getting audio stream: NotReadableError: The I/O read operation failed.
	(anonymous function) (talk-main.js:1:7086944)
	promiseReactionJob
```

</details>

## Server configuration

**Operating system**: Ubuntu/RedHat/...

**Web server:** Apache/Nginx

**Database:** MySQL/Maria/SQLite/PostgreSQL

**PHP version:** 7.2/7.3/7.4

**Nextcloud Version:** (see admin page)

**List of activated apps:**

<details>

```
```
</details>

**Nextcloud configuration:**

<details>

```
```
</details>

### Server log (data/nextcloud.log)
<details>

```
Insert your server log here
```
</details>

**Comment by nickvergessen** [comment]
cc @danxuliu maybe you can have a look on monday

**Comment by nickvergessen** [comment]
I guess this should work now since:
* #6615 
* #6609
* #6601

If not feel free to reopen a new issue

**Comment by danxuliu** [comment]
Note that the are still strange issues when using Safari with Airpods, like the local video becoming black when you remove the Airpods, or audio stopping without the possibility to enable it again. Those issues are from Safari itself, and unfortunately, as far as I know, it is not possible to workaround them in Talk until they are properly fixed in Safari.
- There is no voice when restarting airpods using webrtc in safari https://bugs.webkit.org/show_bug.cgi?id=189645
- Airpods and other bluetooth headphones causes audio to stop during webrtc calls https://bugs.webkit.org/show_bug.cgi?id=213854
- A MediaStreamTrack ended due to a capture failure - When using Airpods https://bugs.webkit.org/show_bug.cgi?id=232835
- Safari local video element pauses after bluetooth audioinput is disconnected https://bugs.webkit.org/show_bug.cgi?id=231787
- MediaDevices devicechange event loop https://bugs.webkit.org/show_bug.cgi?id=226921

**Comment by nickvergessen** [comment]
Extracted to a new issue so search results find an open ticket to it.
